torch/testing/_internal/common_fsdp.py

```diff
@@ -1324,8 +1332,32 @@ def fully_shard_with_compiled_compute(*args, **kwargs):
         if compile_compute_on_module is None or isinstance(
             args[0], compile_compute_on_module
         ):
+            torch.distributed._composable.fsdp.fully_shard(*args, **kwargs)  # type: ignore[operator]
+            for handle_id, hook in args[0]._forward_pre_hooks.items():
+                if (
+                    hook.__module__ == "torch.distributed._composable.fsdp._fsdp_state"
+                    and hook.__name__ == "_pre_forward"
+                ):
+                    args[0]._forward_pre_hooks[handle_id] = torch.compiler.disable(hook)
```

Comment:
I wonder if we should land this `torch.compiler.disable()` for now. Without this, we may error during tracing the forward pre-hook in some cases (e.g. 2D with `DTensor`).

cc: @yf225

Reply:
will split out `torch.compile.disable` into a separate PR to discuss with @yf225